docs/source/en/tasks/object_detection.mdx

```diff
@@ -0,0 +1,487 @@
+    def __init__(self, img_folder, feature_extractor, ann_file):
+        super(CocoDetection, self).__init__(img_folder, ann_file)
+        self.feature_extractor = feature_extractor
+
+    def __getitem__(self, idx):
+        # read in PIL image and target in COCO format
+        img, target = super(CocoDetection, self).__getitem__(idx)
+
+        # preprocess image and target (converting target to DETR format, resizing + normalization of both image and target)
+        image_id = self.ids[idx]
+        target = {"image_id": image_id, "annotations": target}
+        encoding = self.feature_extractor(images=img, annotations=target, return_tensors="pt")
+        pixel_values = encoding["pixel_values"].squeeze()  # remove batch dimension
+        target = encoding["labels"][0]  # remove batch dimension
+
+        return {"pixel_values": pixel_values, "labels": target}
+
+
+im_processor = DetrImageProcessor.from_pretrained("./detr-resnet-50_fine_tuned_cppe5")
+model = DetrForObjectDetection.from_pretrained("./detr-resnet-50_fine_tuned_cppe5")
+
+# prepare the test dataset
+def val_formatted_anns(image_id, objects):
+    annotations = []
+    for i in range(0, len(objects["id"])):
+        new_ann = {
+            "id": objects["id"][i],
+            "category_id": objects["category"][i],
+            "iscrowd": 0,
+            "image_id": image_id,
+            "area": objects["area"][i],
+            "bbox": objects["bbox"][i],
+        }
+        annotations.append(new_ann)
+
+    return annotations
+
+
+# CocoDetection expects to read the data from files in a certain format
+def save_cppe5_annotation_file_images(cppe5):
+    output_json = {}
+    path_output_cppe5 = f"{os.getcwd()}/cppe5/"
+
+    if not os.path.exists(path_output_cppe5):
+        os.makedirs(path_output_cppe5)
+
+    path_anno = os.path.join(path_output_cppe5, "cppe5_ann.json")
+    categories_json = [{"supercategory": "none", "id": id, "name": id2label[id]} for id in id2label]
+    output_json["images"] = []
+    output_json["annotations"] = []
+    for batch in cppe5:
+        ann = val_formatted_anns(batch["image_id"], batch["objects"])
+        output_json["images"].append(
+            {
+                "id": batch["image_id"],
+                "width": batch["image"].width,
+                "height": batch["image"].height,
+                "file_name": f"{batch['image_id']}.png",
+            }
+        )
+        output_json["annotations"].extend(ann)
+    output_json["categories"] = categories_json
+
+    with open(path_anno, "w") as file:
+        json.dump(output_json, file, ensure_ascii=False, indent=4)
+
+    for im, img_id in zip(cppe5["image"], cppe5["image_id"]):
+        path_img = os.path.join(path_output_cppe5, f"{img_id}.png")
+        im.save(path_img)
+
+    return path_output_cppe5, path_anno
+
+
+path_output_cppe5, path_anno = save_cppe5_annotation_file_images(cppe5["test"])
+
+dummy = CocoDetection(path_output_cppe5, im_processor, path_anno)
```

Comment:
I don't think it's a dummy dataset. 

Suggestion: `test_ds_coco_format`.

Reply:
Fixed in 1a38260b27ef977e0d82ba1c918eeb19c8f3dcb7